### DISCUSSION TITLE
The last version of VS Code does'nt work for debugging in Ruby

### DISCUSSION THREAD

**Discussion by sallisa** [microsoft/vscode-discussions#1230] [Extension Development QnA]
``I tried to start a debugging session for a code developed in Ruby by using  the launch request : 

```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Debug TestEnv",
            "type": "Ruby",
            "request": "launch",
            "program": "${workspaceRoot}/testEnv.rb",
            "showDebuggerOutput": true,
            "cwd": "${workspaceRoot}",
            "args": [],
            "env": {
                "RUBY_DEBUG_HOST": "127.0.0.1",
                "RUBY_DEBUG_PORT": "1234"
            }
        }
    ]
}
```

In the DEBUG CONSOLE, I receive the error message :

**Debugger (ruby-debug-ide 0.7.3, debase 0.2.4.1, file filtering is supported) listens on 127.0.0.1:1234

Debugger error: Client: Error: connect ECONNREFUSED ::1:1234** 

The ruby extension is installed.
I have the correct versions of ruby-debug-ide and debase installed

`ruby-debug-ide (0.7.3)
debase (0.2.4.1)
`
I tried to start the debug server manually : 
`rdebug-ide --host 127.0.0.1 --port 1234 testEnv.rb
`

and to listen for rdebug-ide by using  the attach request : 
`{
            "name": "Listen for rdebug-ide",
            "type": "Ruby",
            "request": "attach",
            "cwd": "${workspaceRoot}",
            "remoteHost": "127.0.0.1",
            "remotePort": "1234",
            "remoteWorkspaceRoot": "${workspaceRoot}",
            "showDebuggerOutput": true,
        }`
and it works but I want to use the launch request directly and debug in the DEBUG CONSOLE.

**Comment by isidorn** [comment]
Please file this issue here https://github.com/Shopify/ruby-lsp/tree/main/vscode
And you should use the ruby extension from Shopify. Not the outdated one.
